File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let output = 0
    for(let i = 0; i < s.length; i++) {
        let char = s[i]
        for(let j = i + 1; j < s.length; j++) {
            if(char.indexOf(s[j]) !== -1) break
            else char += s[j]
        }
        output = Math.max(output, char.length)
    }
    return output
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if(s.length() == 0) return 0;
        HashMap<Character, Integer> mapped = new HashMap<>();
        int max = 0;
        for(int i = 0, j = 0; i < s.length(); i++) {
            if(mapped.containsKey(s.charAt(i))) {
                j = Math.max(j, mapped.get(s.charAt(i)) + 1);
            }
            mapped.put(s.charAt(i), i);
            max = Math.max(max, i - j + 1);
        }
        return max;
    }
}